pkg/gce-cloud-provider/compute/gce.go

```diff
@@ -441,3 +442,27 @@ func IsGCENotFoundError(err error) bool {
 func IsGCEInvalidError(err error) bool {
 	return IsGCEError(err, "invalid")
 }
+
+// ErrorContainsRegex checks if the error message contains a specific regex pattern.
+func ErrorContainsRegex(err error, regexPattern string) bool {
+	if err == nil {
+		return false
+	}
+	re, compileErr := regexp.Compile(regexPattern)
+	if compileErr != nil {
+		fmt.Printf("Error compiling regex '%s': %v\n", regexPattern, compileErr)
+		return false
+	}
+	errorMessage := err.Error()
+	return re.MatchString(errorMessage)
+}
+
+// IsSnapshotAlreadyExistsError checks if the error is a snapshot already exists error.
+func IsSnapshotAlreadyExistsError(err error) bool {
+	return ErrorContainsRegex(err, "The resource '[^']+' already exists")
+}
+
+// IsGCPOrgViolationError checks if the error is a GCP organization policy violation error.
+func IsGCPOrgViolationError(err error) bool {
+	return ErrorContainsRegex(err, "violates constraint constraints/gcp.")
```

Comment:
Is there a more complex error policy violation field in the response that you can check instead of string error message extraction?

The risk is if the error message from the API is reformatted, this regex matching may fail. If there is a more dependable API type response we could use, I would prefer to use that.